### ISSUE TITLE
Implicit declaration of function 'SYNC_LOG_OBJC_MAYBE' is invalid in C99

### ISSUE THREAD

**Issue by andpor** [CocoaLumberjack/CocoaLumberjack#757]
### New Issue Checklist
- [x] I have read and understood the [CONTRIBUTING guide](https://github.com/CocoaLumberjack/CocoaLumberjack/blob/master/.github/CONTRIBUTING.md)
- [x] I have read the [Documentation](https://github.com/CocoaLumberjack/CocoaLumberjack#documentation) or [CocoaDocs](cocoadocs.org/docsets/CocoaLumberjack)
- [x] I have searched for a similar issue in the [project](https://github.com/CocoaLumberjack/CocoaLumberjack/issues) and found none
### Issue Info

| Info | Value |
| --- | --- |
| Platform Name | ios |
| Platform Version | 8.0 |
| CocoaLumberjack Version | 2.3.0 |
| Integration Method | manually |
| Xcode Version | Xcode 7.3 |
| Repro rate | 100% |
| Demo project link | description provided |
### Issue Description and Steps

During compilation error is generated:
```
myfile.m:83:13: Implicit declaration of function 'SYNC_LOG_OBJC_MAYBE' is invalid in C99
```

This happens at macro expansion for MyDebug

We have our own macro defined that looks like this:
```objc
# define  MyDebug(level, s, ... )  \

do { if(level & LOG_FLAG_FATAL || level & LOG_FLAG_ERROR) SYNC_LOG_OBJC_MAYBE(LOG_LEVEL_DEF, level, CTXGENERAL, s, ##__VA_ARGS__); else ASYNC_LOG_OBJC_MAYBE(LOG_LEVEL_DEF, level, CTXGENERAL, s, ##__VA_ARGS__); } while(0)
```

This was working fine when we were using version 2.0 with the raw file for Lumberjack included in the project. Now we are using an embedded fmk generated from PubNub build.

Also, there used to be macros in Luberjack for C. Example `SYNC_LOG_C_MAYBE` or  `ASYNC_LOG_C_MAYBE`. What happened to them? Can you point me to documentation explaining how to migrate?

**Comment by bpoplauschi** [comment]
Probably another instance of the deprecation of legacy macros @CocoaLumberjack/collaborators

**Comment by hhanesand** [comment]
> Also, there used to be macros in Luberjack for C. Example SYNC_LOG_C_MAYBE or ASYNC_LOG_C_MAYBE. What happened to them? Can you point me to documentation explaining how to migrate?
@andpor Replace all the C macros with their normal equivalent. See #220 #229 

`SYNC_LOG_OBJC_MAYBE` -> `LOG_MAYBE(NO, ...` (no means synchronous)
`ASYNC_LOG_C_MAYBE` -> `LOG_MAYBE(YES, ...`

See `DDLogMacros.h` for the macros we provide.

**Comment by sushichop** [comment]
Hi, guys!
I have updated demo and docs about CustomLogLevels. See #1023.
@CocoaLumberjack/collaborators

**Comment by sushichop** [comment]
@bpoplauschi Yes. We can close this.

**Comment by sushichop** [comment]
Fixed by #1023
